core/windows_and_tabs/window_snap.py

```diff
@@ -345,22 +362,22 @@ def snap_app(app_name: str, position: RelativeScreenPos):
         _bring_forward(window)
         _snap_window_helper(window, position)
 
-    def snap_layout(
-        positions_by_count: Dict[int, list[RelativeScreenPos]],
+    def snap_app_layout(
+        positions: list[RelativeScreenPos],
         apps: list[str],
     ):
         """Split the screen between multiple applications."""
-        try:
-            positions = positions_by_count[len(apps)]
-        except KeyError:
-            supported_layouts = ", ".join(map(str, positions_by_count.keys()))
-            message = f"{len(apps)} applications given but chosen layout only supports {supported_layouts}"
-            actions.app.notify(message, "Cannot arrange")
-            raise NotImplementedError(message)
-        for index, app in enumerate(apps):
-            window = _get_app_window(app)
-            _snap_window_helper(window, positions[index])
-            window.focus()
+        windows = [_get_app_window(app) for app in apps]
+        _snap_layout(positions, windows)
+
+    def snap_window_layout(
+        positions: list[RelativeScreenPos],
+        num_windows: int,
+    ):
+        """Snap next layout using the top three application windows"""
+        top_n_windows = ui.windows()[:num_windows]
```

Comment:
Some of these windows aren't snappable-that can result in an error. I don't know how to filter such windows here though.